spring-batch-docs/asciidoc/readersAndWriters.adoc

```diff
@@ -3282,43 +2784,33 @@ public class CustomItemWriter<T> implements ItemWriter<T> {
 ----
 
 [[restartableWriter]]
-
-
 ===== Making the ItemWriter Restartable
 
-To make the `ItemWriter` restartable, we would follow the same
-process as for the `ItemReader`, adding and
-implementing the `ItemStream` interface to
-synchronize the execution context. In the example, we might have to
-count the number of items processed and add that as a footer record.
-If we needed to do that, we could implement
-`ItemStream` in our
-`ItemWriter` so that the counter was
-reconstituted from the execution context if the stream was
-re-opened.
-
-In many realistic cases, custom `ItemWriters` also delegate to
-another writer that itself is restartable (for example, when writing to a
-file), or else it writes to a transactional resource and so does not need
-to be restartable, because it is stateless. When you have a stateful
-writer you should probably be sure to implement
-`ItemStream` as well as
-`ItemWriter`. Remember also that the client of
-the writer needs to be aware of the `ItemStream`,
-so you may need to register it as a stream in the configuration
-XML.
+To make the `ItemWriter` restartable, we would follow the same process as for the
+`ItemReader`, adding and implementing the `ItemStream` interface to synchronize the
+execution context. In the example, we might have to count the number of items processed
+and add that as a footer record. If we needed to do that, we could implement
+`ItemStream` in our `ItemWriter` so that the counter was reconstituted from the execution
+context if the stream was re-opened.
+
+In many realistic cases, custom `ItemWriters` also delegate to another writer that itself
+is restartable (for example, when writing to a file), or else it writes to a
+transactional resource and so does not need to be restartable, because it is stateless.
+When you have a stateful writer you should probably be sure to implement `ItemStream` as
+well as `ItemWriter`. Remember also that the client of the writer needs to be aware of
+the `ItemStream`, so you may need to register it as a stream in the configuration XML.
```

Comment:
We can drop XML.   configuration is enough.

Reply:
Thanks some more, Glenn. :)